arangod/Cluster/ClusterInfo.cpp

```diff
@@ -261,6 +264,131 @@ inline arangodb::AgencyOperation CreateCollectionSuccess(
                                    info};
 }
 
+// make sure a collection is still in Plan
+// we are only going from *assuming* that it is present
+// to it being changed to not present.
+class CollectionWatcher
+    : public std::enable_shared_from_this<CollectionWatcher> {
+ public:
+  CollectionWatcher(CollectionWatcher const&) = delete;
+  CollectionWatcher(AgencyCallbackRegistry* agencyCallbackRegistry,
+                    LogicalCollection const& collection);
+  ~CollectionWatcher();
+
+  bool isPresent() {
+    // Make sure we did not miss a callback
+    _agencyCallback->refetchAndUpdate(true, false);
+    return _present.load();
+  }
+
+ private:
+  AgencyCallbackRegistry* _agencyCallbackRegistry;
+  std::shared_ptr<AgencyCallback> _agencyCallback;
+
+  // TODO: this does not really need to be atomic: We only write to it
+  //       in the callback, and we only read it in `isPresent`; it does
+  //       not actually matter whether this value is "correct".
+  std::atomic<bool> _present;
+};
+
+constexpr frozen::unordered_map<std::string_view, ServerHealth, 4>
+    kHealthStatusMap = {
+        {consensus::Supervision::HEALTH_STATUS_BAD, ServerHealth::kBad},
+        {consensus::Supervision::HEALTH_STATUS_FAILED, ServerHealth::kFailed},
+        {consensus::Supervision::HEALTH_STATUS_GOOD, ServerHealth::kGood},
+        {consensus::Supervision::HEALTH_STATUS_UNCLEAR,
+         ServerHealth::kUnclear}};
+
+[[nodiscard]] ServersKnown parseServersKnown(
+    VPackSlice const serversKnownSlice, VPackSlice const supervisionHealth,
+    containers::FlatHashSet<ServerID> const& serverIds) {
+  ServersKnown res;
+  TRI_ASSERT(serversKnownSlice.isNone() || serversKnownSlice.isObject());
+  LOG_TOPIC("91da8", TRACE, Logger::CLUSTER)
+      << "Supervision health is:" << supervisionHealth.toString();
+  if (serversKnownSlice.isObject()) {
+    for (auto const it : VPackObjectIterator(serversKnownSlice)) {
+      auto status = ServerHealth::kUnclear;
+      std::string serverId = it.key.copyString();
```

Comment:
```suggestion
      std::string_view serverId = it.key.stringView();
```

Reply:
We anyway will need a string copy here. As at line 336 we will move this string into result. So we can have a string_view here, but still later we will copy it into result.  Wit that assumption I think copyString here is fine.

Reply:
Ok